apache/superset
# fix(explore): Chart save modal displays error instead of failing silently

### SUMMARY
When user tried to save a chart and a request executed by SaveModal chart failed, the modal would just close and there would be no indication that an error occurred. That might lead the user to think that their chart was successfully saved even though it wasn't.
This PR fixes that behaviour by not closing the modal and displaying an alert with error message within it. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

https://user-images.githubusercontent.com/15073128/197514948-a996021b-bd5d-4939-aee5-c9f23c34b2d6.mov

After:

https://user-images.githubusercontent.com/15073128/197514625-2667b4fe-519b-4313-be65-58e00a31c66f.mov

### TESTING INSTRUCTIONS
1. Open the chart save modal
2. Turn off the Internet (to trigger an error)
3. Click save button 
4. See that the error is displayed and the modal doesn't close

### ADDITIONAL INFORMATION



CC @codyml
